zerver/middleware.py

```diff
@@ -368,7 +369,7 @@ def process_request(self, request: HttpRequest) -> None:
     def process_view(
         self,
         request: HttpRequest,
-        view_func: ViewFuncT,
+        view_func: Callable[Concatenate[HttpRequest, ParamT], HttpResponse],
         args: List[str],
```

Comment:
This middleware has to work with every view, so `view_func` should return `HttpResponseBase`, and `args` are [not necessarily `str`](https://docs.djangoproject.com/en/4.1/topics/http/urls/#path-converters).